csatar-plugins/csatar/components/OrganizationUnitFrontend.php

```diff
@@ -127,4 +134,206 @@ public function onCancelEdit()
+
+        if (Input::get('cancel')) {
+            $this->page['showUploadForm'] = false;
+        }
+
+        return [
+            '#uploadCsv' => $this->renderPartial('@csvUploadForm')
+        ];
+    }
+
+    public function onImportScoutsToCsv(){
+        $file = Input::file('csvFile');
+        $teamId = Input::get('teamId');
+
+        if (empty($file) || empty($teamId)) {
+            return; //TODO: flash here
+        }
+
+        if ($file->isValid()) {
+            $file = $file->move(temp_path(), $file->getClientOriginalName());
+        }
+
+        $csvData = [];
+        if (($handle = fopen($file, "r")) !== FALSE) {
+            while (($data = fgetcsv($handle)) !== FALSE) {
+                $csvData[] = $data;
+            }
+        }
+
+        $attributes = $csvData[0];
+        $legalRelationshipsMap = (new LegalRelationshipMapper)->namesToIds;
+        $religionsMap = (new ReligionMapper)->namesToIds;
+        $log = [];
+
+        foreach ($csvData as $rowNumber => $rowData) {
+            if ($rowNumber == 0 || $rowNumber == 1) {
+                continue;
+            }
+            $personalIdentificationNumber = $rowData[array_search('personal_identification_number', $attributes)];
+            $ecsetCode = $rowData[array_search('ecset_code', $attributes)];
+
+            $data = [];
+            foreach ($rowData as $key => $value) {
+                if ($attributes[$key] == 'gender') {
+                    $data[$attributes[$key]] = array_flip(Gender::getOptionsWithLabels())[$value];
+                    continue;
+                }
+                if ($attributes[$key] == 'legal_relationship_id') {
+                    $data[$attributes[$key]] = $legalRelationshipsMap[$value];
+                    continue;
+                }
+                if ($attributes[$key] == 'religion_id') {
+                    $data[$attributes[$key]] = $religionsMap[$value];
+                    continue;
+                }
+                $data[$attributes[$key]] = $value;
+            }
+
+            $firstOrNewConditions = [
+                'team_id' => $teamId,
+            ];
+            unset($data['team_id']);
+            unset($data['ecset_code']);
+
+            if (empty($personalIdentificationNumber) && !empty($ecsetCode)) {
+                $firstOrNewConditions['ecset_code'] = $ecsetCode;
+            }
+
+            if (empty($ecsetCode) && !empty($personalIdentificationNumber)) {
+                $firstOrNewConditions['personal_identification_number'] = $personalIdentificationNumber;
+                unset($data['personal_identification_number']);
+            }
+
+            if (!empty($ecsetCode) && !empty($personalIdentificationNumber)) {
+                $firstOrNewConditions['personal_identification_number'] = $personalIdentificationNumber;
+                $firstOrNewConditions['ecset_code'] = $ecsetCode;
+                unset($data['personal_identification_number']);
+            }
```

Comment:
Ez a 3 if miert kell ilyen bonyolult legyen? Nem csak annyi tortenik, hogy ha az egyik nem ures, akkor beallitjuk, s ha a masik nem ures, akkor azt is beallitjuk?

Reply:
Igaz